nav2_core/include/nav2_core/navigator.hpp

```diff
@@ -106,23 +106,121 @@ class NavigatorMuxer
   std::mutex mutex_;
 };
 
+class NavigatorBase
+{
+public:
+  RCLCPP_SMART_PTR_DEFINITIONS(NavigatorBase)
+  NavigatorBase()
+  {
+    plugin_muxer_ = nullptr;
+  }
+
+  virtual bool on_configure(
+    rclcpp_lifecycle::LifecycleNode::WeakPtr parent_node,
+    const std::vector<std::string> &,
+    const FeedbackUtils & feedback_utils,
+    nav2_core::NavigatorMuxer * plugin_muxer,
+    std::shared_ptr<nav2_util::OdomSmoother>)
+  {
+    auto node = parent_node.lock();
+    RCLCPP_INFO(node->get_logger(), "Configuring %s", getName().c_str());
+    
+    logger_ = node->get_logger();
+    clock_ = node->get_clock();
+    feedback_utils_ = feedback_utils;
+    plugin_muxer_ = plugin_muxer;
+
+    // get the default behavior tree for this navigator
+    default_bt_xml_filename_ = getDefaultBTFilepath(parent_node);
+    return true;
+  }
+
+  /**
+   * @brief Get the action name of this navigator to expose
+   * @return string Name of action to expose
+   */
+  virtual std::string getName() = 0;
+
+  virtual std::string getDefaultBTFilepath(rclcpp_lifecycle::LifecycleNode::WeakPtr node) = 0;
+  
+  /**
+   * @brief Activation of the navigator's backend BT and actions
+   * @return bool If successful
+   */
+  virtual bool on_activate()
+  {
+    return activate();
+  }
+
+  /**
+   * @brief Deactivation of the navigator's backend BT and actions
+   * @return bool If successful
+   */
+  virtual bool on_deactivate()
+  {
+    return deactivate();
+  }
+
+  /**
+   * @brief Cleanup a navigator
+   * @return bool If successful
+   */
+  virtual bool on_cleanup()
+  {
+    return cleanup();
+  }
+
+protected:
+
+  /**
+   * @param Method to configure resources.
+   */
+  virtual bool configure(
```

Comment:
Change all these to pure virtuals (e.g. no implementation `= 0`) as required implementations. Configure/cleanup/activate/deactivate